src/Codec/Archive/Zip.hs

```diff
@@ -459,16 +480,16 @@ addFilesToArchive opts archive files = do
 extractFilesFromArchive :: [ZipOption] -> Archive -> IO ()
 extractFilesFromArchive opts archive = do
   let entries = zEntries archive
-  if OptPreserveSymbolicLinks `elem` opts
-    then do
+  case [s | OptSymbolicLinks s <- opts] of
+    _ : _ -> do
 #ifdef _WINDOWS
       mapM_ (writeEntry opts) entries
 #else
       let (symbolicLinkEntries, nonSymbolicLinkEntries) = partition isEntrySymbolicLink entries
       mapM_ (writeEntry opts) nonSymbolicLinkEntries
       mapM_ (writeSymbolicLinkEntry opts) symbolicLinkEntries
```

Comment:
I think it would be cleaner to modify `writeEntry` so that it checks to see if the entry is a symbolic link and then behaves appropriately depending on the options.  Doing this partition seems awkward to me, it means that we're checking twice to see if the entry is a symbolci link (there's another check in writeSymbolicLinkEntry).

Reply:
done, only checks once via writeEntry